ee/api/billing.py

```diff
@@ -87,46 +87,67 @@ def patch(self, request: Request, *args: Any, **kwargs: Any) -> Response:
 
         return self.list(request, *args, **kwargs)
 
+    class ActivationSerializer(serializers.Serializer):
+        plan = serializers.CharField(required=False)
+        products = serializers.CharField(
+            required=False
+        )  # This is required but in order to support an error for the legacy 'plan' param we need to set required=False
+        redirect_path = serializers.CharField(required=False)
+
+        def validate(self, data):
+            plan = data.get("plan")
+            products = data.get("products")
+
+            if plan and not products:
+                raise ValidationError(
+                    {
+                        "plan": "The 'plan' parameter is no longer supported. Please use the 'products' parameter instead."
+                    }
+                )
+            if not products:
+                raise ValidationError({"products": "The 'products' parameter is required."})
+
+            return data
+
     @action(methods=["GET"], detail=False)
     def activation(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
         license = get_cached_instance_license()
         organization = self._get_org_required()
 
-        redirect_path = request.GET.get("redirect_path") or "organization/billing"
+        serializer = self.ActivationSerializer(data=request.GET)
```

Comment:
Rename to `Activate` since you did that in Billing?

Reply:
Yep, that'll be on another PR when I move the URL in posthog as well. Doing that after these others are merged.